dashboards-observability/public/components/visualizations/charts/bar/bar.tsx

```diff
@@ -10,80 +10,174 @@ import { LONG_CHART_COLOR, PLOTLY_COLOR } from '../../../../../common/constants/
-  const yaxis =
-    dataConfig.valueOptions && dataConfig.valueOptions.xaxis ? dataConfig?.valueOptions.yaxis : [];
+  const dataConfigTab =
+    visualizations.data?.rawVizData?.bar?.dataConfig &&
+    visualizations.data.rawVizData.bar.dataConfig;
+  const xaxis = dataConfigTab?.dimensions
+    ? dataConfigTab?.dimensions.filter((item) => item.label)
+    : [];
+  const yaxis = dataConfigTab?.metrics ? dataConfigTab?.metrics.filter((item) => item.label) : [];
   const barOrientation = dataConfig?.chartStyles?.orientation || vis.orientation;
-  const { defaultAxes } = visualizations.data;
-  const tickAngle = dataConfig?.chartStyles?.rotateBarLabels || vis.labelAngle
+  const tickAngle = dataConfig?.chartStyles?.rotateBarLabels || vis.labelAngle;
   const lineWidth = dataConfig?.chartStyles?.lineWidth || vis.lineWidth;
-  const fillOpacity = dataConfig?.chartStyles?.fillOpacity !== undefined ? dataConfig?.chartStyles?.fillOpacity / FILLOPACITY_DIV_FACTOR : vis.fillOpacity / FILLOPACITY_DIV_FACTOR;
+  const fillOpacity =
+    dataConfig?.chartStyles?.fillOpacity !== undefined
+      ? dataConfig?.chartStyles?.fillOpacity / FILLOPACITY_DIV_FACTOR
+      : vis.fillOpacity / FILLOPACITY_DIV_FACTOR;
   const barWidth = 1 - (dataConfig?.chartStyles?.barWidth || vis.barWidth);
   const groupWidth = 1 - (dataConfig?.chartStyles?.groupWidth || vis.groupWidth);
   const isVertical = barOrientation === vis.orientation;
-  const showLegend = !(dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== vis.showLegend);
+  const showLegend = !(
+    dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== vis.showLegend
+  );
   const legendPosition = dataConfig?.legend?.position || vis.legendPosition;
+  visualizations.data?.rawVizData?.dataConfig?.metrics
+    ? visualizations.data?.rawVizData?.dataConfig?.metrics
+    : [];
+  const labelSize = dataConfig?.chartStyles?.labelSize || 14;
+  let bars, valueSeries, valueForXSeries;
 
-  const getSelectedColorTheme = (field: any, index: number) => dataConfig?.colorTheme?.length > 0 && dataConfig.colorTheme.find(
-    (colorSelected) => colorSelected.name.name === field.name)?.color || PLOTLY_COLOR[index % PLOTLY_COLOR.length];
+  const getSelectedColorTheme = (field: any, index: number) =>
+    (dataConfig?.colorTheme?.length > 0 &&
+      dataConfig.colorTheme.find((colorSelected) => colorSelected.name.name === field.label)
+        ?.color) ||
+    PLOTLY_COLOR[index % PLOTLY_COLOR.length];
 
-  let valueSeries;
   if (!isEmpty(xaxis) && !isEmpty(yaxis)) {
     valueSeries = isVertical ? [...yaxis] : [...xaxis];
+    valueForXSeries = isVertical ? [...xaxis] : [...yaxis];
   } else {
-    valueSeries = defaultAxes.yaxis || take(fields, lastIndex > 0 ? lastIndex : 1);
+    return <EmptyPlaceholder icon={visualizations?.vis?.iconType} />;
   }
+  const prepareData = (valueForXSeries) => {
+    return [...valueForXSeries.map((dimension: any) => data[dimension.label])]?.reduce(function (
```

Comment:
map and reduce are going to return a new array anyway that does not alter `valueForXSeries`. So what it the purpose of creating another array using destructuring operator and creating another array from that before you return?